### DISCUSSION TITLE
ArrowCircleTip and ArrowCircleFilledTip isn't working

### DISCUSSION THREAD

**Discussion by mscharling** [3b1b/manim#1597] [Q&A] [answered]
Hi, 

I'm trying to run the following code:

```
class int1(Scene):
    CONFIG={
        # "camera_config":{"background_color":"#FFFFFF"}
        "label_nums_color":{BLACK},
        "number_scale_val":{0.1}
    }

    def construct(self):
        numberline_kwargs = {
            "x_min": -10,
            "x_max": 10+1,
            "unit_size": 0.62,
            "include_numbers": True,
            "include_tip": True,
            "number_scale_val": 0.5
        }

        l0 = NumberLine(**numberline_kwargs).set_color(color = BLACK)

        Aleft = ArrowCircleTip([2, 0, 0], [-2, 0, 0], color = GREEN)
        Aleft.move_to(.62*1.3*UP+.62*4*LEFT)

        Aright = ArrowCircleFilledTip([2, 0, 0], [3, 0, 0], color = GREEN)
        Aright.move_to(.62*1.3*UP+.62*1*RIGHT)

        self.add(l0, Aleft, Aright)

```
And I get a long list of errors in my command prompt when running the code. It seems to me that the objects arent defined in manimlib somehow - but I'm not that advanced to find the correct solution. How can I define the objects? Or what am I missing?

**Comment by TonyCrane** [comment, marked as answer]
In manimgl, there is no `ArrowCircleTip` or `ArrowCircleFilledTip`. But they are in the community version of manim: [ManimCommunity/manim](https://github.com/ManimCommunity/manim).

And `ArrowCircleTip` and `ArrowCircleFilledTip` are just circles, I guess you want to use like `Arrow([2, 0, 0], [-2, 0, 0], tip_shape=ArrowCircleTip)`. And in manimgl, you can get the same result by using `Line([2, 0, 0], [-2, 0, 0]).add_tip(tip_style=2)`

**Reply by TonyCrane** [reply]
For more about `tip_style`, see: #1538
